### ISSUE TITLE
generate_trajectories getting stuck

### ISSUE THREAD

**Issue by RishiHazra** [askforalfred/alfred#120]
Hi @MohitShridhar 

I'm trying to generate new trajectories. However, the generation always gets stuck after resetting the environment: Here is output when I run it:

```
Force Unsave Data: False
Loaded 0 known failed tuples
Found path: ...
Mono path[0] = ...
Mono config path = ...
Preloaded 'ScreenSelector.so'
Display 0 '0': 1920x1200 (primary device).
Logging to ...
ThorEnv started.
###################################

Successes: 
Empty DataFrame
Columns: [goal, pickup, movable, receptacle, scene]
Index: []

##################################
('pick_and_place_with_movable_recep', 'Fork', 'Cup', 'DiningTable', 17)
sampled tuple: ('pick_and_place_with_movable_recep', 'Fork', 'Cup', 'DiningTable', 17)
PDDLGoalType: pick_and_place_with_movable_recep
Saving images to: dataset/new_trajectories/pick_and_place_with_movable_recep-Fork-Cup-DiningTable-17/trial_T20220806_035502_492636/raw_images/
Resetting ThorEnv
```
I have installed the exact version of AI2Thor as mentioned in the requirements.txt file. After debugging for a while, I was able to narrow it down to https://github.com/askforalfred/alfred/blob/1898b83547b589b8635737929e04ee4f2f404177/env/thor_env.py#L66 
However, I'm unable to debug it. Would appreciate some help here. 

Thanks in advance !!!

**Comment by askforalfred** [comment]
Can you check the error in "/home/rishihazra/.config/unity3d/Allen
Institute for Artificial Intelligence/AI2-Thor/Player.log" ?

**Comment by RishiHazra** [comment]
```
Fallback handler could not load library /home/rishihazra/.ai2thor/releases/thor-Linux64-bdcefe04c17bef073ecfe3d90786e84740f7addf/thor-Linux64-bdcefe04c17bef073ecfe3d90786e84740f7addf_Data/Mono/libadvapi32.dll.so
Fallback handler could not load library /home/rishihazra/.ai2thor/releases/thor-Linux64-bdcefe04c17bef073ecfe3d90786e84740f7addf/thor-Linux64-bdcefe04c17bef073ecfe3d90786e84740f7addf_Data/Mono/libadvapi32.dll
```

**Comment by askforalfred** [comment]
That particular error feels outside the scope of the ALFRED codebase and is
instead within the AI2-THOR installation.

**Comment by RishiHazra** [comment]
I suspected that. Not sure why though. Anyways, appreciate the quick response. Thanks!

**Comment by YaqiXie** [comment]
@RishiHazra Did you solve the problem? I got the same issue Orz

**Comment by RishiHazra** [comment]
> @RishiHazra Did you solve the problem? I got the same issue Orz

Hi, the problem was mainly because of the torch module. You could try two things: The AI2Thor recommended version + older version of torch (https://github.com/alexpashevich/E.T./issues/8) or upgrading the AI2Thor package + newer torch version

**Comment by YaqiXie** [comment]
@RishiHazra I tried to use the recommended version torch 1.1.0 and torchvision 0.3.0. But it still doesn't work. And even tried a virtual env without torch at all. But I still run into the same issue Orz No idea why.

**Comment by RishiHazra** [comment]
> @RishiHazra I tried to use the recommended version torch 1.1.0 and torchvision 0.3.0. But it still doesn't work. And even tried a virtual env without torch at all. But I still run into the same issue Orz No idea why.

@YaqiXie: In my case, I had to upgrade the AI2Thor version (ai2thor~=4.2.0) to be compatible with my torch version (torch==1.11.0+cu113 torchvision==0.12.0+cu113 torchaudio==0.11.0). However, note that you might have to make slight changes in the ALFRED code too since it is based on an older version of AI2Thor (2.1.0). I'd be happy to help with that.

Also, see if you can find something useful in another similar issue (https://github.com/alexpashevich/E.T./issues/8) if you want to keep using the recommended version.

**Comment by YaqiXie** [comment]
@RishiHazra Hi Rishi, Yep, a later version of AI2THOR works for me. How did you modify the ALFRED code? The action API seems different. Thank you a lot for the help!

**Comment by RishiHazra** [comment]
> @RishiHazra Hi Rishi, Yep, a later version of AI2THOR works for me. How did you modify the ALFRED code? The action API seems different. Thank you a lot for the help!

@YaqiXie Hi Yaqi, while I'm unable to share my current code here, I can give you some pointers to help modify the code. Most of the modifications that I made were taken from this repository ([https://github.com/jiasenlu/alfred](https://github.com/jiasenlu/alfred)) which is modified to be compatible with AI2THOR 2.5.1. If I remember correctly, these modifications worked even for AI2THOR 4.2.0. There were some other minor modifications that I had to make in the object poses ([see this](https://github.com/askforalfred/alfred/issues/121)). I'd also recommend you look into the[ AI2THOR documentation of actions](https://ai2thor.allenai.org/ithor/documentation/interactive-physics) to get a clear understanding of what you might need. Feel free to reach out at (rishi.hazra@our.se) in case you run into issues.
